### DISCUSSION TITLE
Child's interactive property should have priority over parent's interactiveChildren

### DISCUSSION THREAD

**Discussion by havrlisan** [pixijs/pixijs#7924] [Q&A]
I'm currently working with a Sprite that has multiple Graphics children, some of which should be interactive and some should not. If you set parent's [interactiveChildren](https://pixijs.download/v6.1.3/docs/PIXI.Sprite.html#interactiveChildren) property to True and specific child's [interactive](https://pixijs.download/v6.1.3/docs/PIXI.Graphics.html#interactive) property to False, you can still interact with it. The same applies to the opposite, if you set interactiveChildren to False, and child's interactive to True, you cannot interact with the child whatsoever.

To be clear, when I say interact, I'm referring to attached mouse events that should get triggered.

Both Sprite and Graphics inherit from Container, so I assume if I were to work with Sprite-Sprite or Graphics-Graphics, I'd get the same results.

**Comment by havrlisan** [comment]
If it helps anyone, my workaround for my specific case (interactiveChildren is true and child's interactive false) is to set child's hitArea to empty polygon (after all other actions):
`someChild.hitArea = new PIXI.Polygon([]);`

**Comment by ivanpopelyshev** [comment]
`interactiveChildren=false` turns off whole subtree, that its purpose.

**Comment by ivanpopelyshev** [comment]
MAybe you want multiple modes for it? not true/false but several, and different behaviours? make your own SearchTree do `renderer.interaction.search = new MySearchTree()`, make correct behaviour, then propose it here.
